Makefile

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+TARGET = federation-v2
+GOTARGET = github.com/kubernetes-sigs/$(TARGET)
+REGISTRY ?= quay.io/kubernetes-multicluster
+IMAGE = $(REGISTRY)/$(TARGET)
+DIR := ${CURDIR}
+DOCKER ?= docker
+
+GIT_VERSION ?= $(shell git describe --always --dirty)
```

Comment:
We should consider adding an `IMAGE_TAG` variable that can be assigned the `GIT_VERSION` by default. This would allow creating an image with any custom tag name the developer chooses.